src/pages/Thanks/ThanksPage.vue

```diff
@@ -105,8 +117,19 @@ export default {
 		}
 	},
 	apollo: {
-		query: checkoutReceiptQuery,
-		preFetch: true,
+		query: thanksPageQuery,
+		preFetch(config, client, { route }) {
+			return client.query({
+				query: thanksPageQuery,
+				variables: {
+					checkoutId: numeral(route.query.kiva_transaction_id).value()
+				}
+			}).then(() => {
+				return client.query({
+					query: experimentAssignmentQuery, variables: { id: 'mg_thanks_cta' }
+				});
+			});
+		},
 		preFetchVariables({ route }) {
```

Comment:
I don't understand why `preFetchVariables` and `variables` is needed in this case, since I'm specifying the variable in the query above...

Reply:
Because `query` is defined in the apollo options, our apollo plugin is calling `preFetchVariables` to use for calling `apollo.readQuery`, and on the client it's calling `variables` to use for calling `apollo.watchQuery` (https://github.com/kiva/ui/blob/master/src/plugins/apollo-plugin.js#L22). If you don't define `query`, those functions won't be needed, but then you also need to setup your own calls to `apollo.readQuery` and `apollo.watchQuery`.